pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -350,6 +350,15 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(ctx context.Context, cl
 			klog.V(4).Infof("synchronizing unbound PersistentVolumeClaim[%s]: no volume found", claimToClaimKey(claim))
 			// No PV could be found
 			// OBSERVATION: pvc is "Pending", will retry
+
+			if utilfeature.DefaultFeatureGate.Enabled(features.RetroactiveDefaultStorageClass) {
+				klog.V(4).Infof("FeatureGate[%s] is enabled, attempting to assign storage class to unbound PersistentVolumeClaim[%s]", features.RetroactiveDefaultStorageClass, claimToClaimKey(claim))
+				if claim, err = ctrl.assignDefaultStorageClass(claim); err != nil {
+					// Intentionally ignoring errors - unsuccessful storage class assignment is expected if there is no default class or PVC already has a storage class
+					klog.V(2).Infof("can't update PersistentVolumeClaim[%s]: %v", claimToClaimKey(claim), err)
```

Comment:
I should have spotted it earlier, claim update error should not be ignored.

Reply:
Done, removing the comment and returning error.